src/api/libcellml/importer.h

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <string>
+
+#include "libcellml/logger.h"
+#include "libcellml/types.h"
+
+namespace libcellml {
+
+/**
+ * @brief The Importer class.
+ *
+ * The Importer class is for representing a CellML Importer.
+ */
+class LIBCELLML_EXPORT Importer: public Logger
+{
+public:
+    ~Importer() override; /**< Destructor */
+    Importer(const Importer &rhs) = delete; /**< Copy constructor */
+    Importer(Importer &&rhs) noexcept = delete; /**< Move constructor */
+    Importer &operator=(Importer rhs) = delete; /**< Assignment operator */
+
+    /**
+     * @brief Create a @c Importer object.
+     *
+     * Factory method to create a @c Importer.  Create a
+     * blank model with::
+     *
+     *   ImporterPtr importer = libcellml::Importer::create();
+     *
+     * @return A smart pointer to a @c Importer object.
+     */
+    static ImporterPtr create() noexcept;
+
+    /**
+     * @brief Flatten the @p model.
+     *
+     * Instantiates all imports and removes them from the @p model.
+     * The result is a self-contained model requiring no external
+     * resources and having no imports.
+     *
+     * @sa clone
+     */
+    ModelPtr flatten(const ModelPtr &inModel);
```

Comment:
Replace "inModel" with "model".

Reply:
This is to distinguish it from the flattened model, which is the variable `model`.

Reply:
Maybe, but it doesn't follow any convention that we have used so far. If anything, I would rename `model` in the body function. Personally, I like to use `res` (for result) for anything that is returned by a function.